### PR TITLE
GLES2 improvements:

### PR DESCRIPTION
**Pull request by aclysma**
- FBOs render with inverted Y so that they can be sampled in the same way as other backends
 - Fix textures bound in descriptor sets using wrong texture unit index
 - Support debug output in GL
 - Demo now uses depth images for the default "invalid" shadow maps

### CODE REVIEW

File: rafx-assets/src/assets/graphics_pipeline/assets.rs [hunk 1]
```diff
@@ -247,7 +247,7 @@ impl MaterialPassData {
                 return Err(error)?;
             }
 
-            log::trace!("  Reflection data:\n{:#?}", reflection_data);
+            //log::trace!("  Reflection data:\n{:#?}", reflection_data);
```
Comment (DavidVonDerau) [inline rafx-assets/src/assets/graphics_pipeline/assets.rs:250]: Commented out?

File: rafx-framework/src/resources/cooked_shader.rs [hunk 2]
```diff
@@ -100,7 +100,7 @@ impl ReflectedShader {
         // We iterate through the entry points we will hit for each stage. Each stage may define
         // slightly different reflection data/bindings in use.
         for reflection_data in entry_points {
-            log::trace!("  Reflection data:\n{:#?}", reflection_data);
+            //log::trace!("  Reflection data:\n{:#?}", reflection_data);
```
Comment (DavidVonDerau) [inline rafx-framework/src/resources/cooked_shader.rs:103]: Also commented.

File: demo-web/shaders/shader.vert.gles [hunk 3]
```diff
@@ -15,5 +15,7 @@ void main()
 {
     interface_var_0 = in_uv;
     gl_Position = vec4(pos, 0.0, 1.0);
```
Comment (DavidVonDerau) [inline demo-web/shaders/shader.vert.gles:17]: What's going on here? Is this how people would have to write GL shaders? Or is this the generated shader code?

**Review by DavidVonDerau — APPROVED** [review]
Just a few questions, otherwise looks good.

File: demo-web/shaders/shader.vert.gles (hunk 3 above)
Comment (aclysma) [reply demo-web/shaders/shader.vert.gles:17]: the shader processor inserts this fix automatically

File: rafx-assets/src/assets/graphics_pipeline/assets.rs (hunk 1 above)
Comment (aclysma) [reply rafx-assets/src/assets/graphics_pipeline/assets.rs:250]: yeah these are super spammy, even for trace

**Pull request merged**